lib/rmt/downloader.rb

```diff
@@ -85,36 +85,26 @@ def verify_checksum(filename, checksum_type, checksum_value)
 
   def make_request(remote_file, local_file, request_fiber, checksum_type = nil, checksum_value = nil)
     uri = URI.join(@repository_url, remote_file)
-    uri.query = @auth_token if @auth_token
+    uri.query = @auth_token if (@auth_token && uri.scheme != 'file')
+
+    if (URI(uri).scheme == 'file' && !File.exist?(uri.path))
+      raise RMT::Downloader::Exception.new("#{remote_file} - File does not exist")
+    end
 
     downloaded_file = Tempfile.new('rmt', Dir.tmpdir, mode: File::BINARY, encoding: 'ascii-8bit')
 
     request = RMT::FiberRequest.new(
       uri.to_s,
       followlocation: true,
       download_path: downloaded_file,
-      request_fiber: request_fiber
+      request_fiber: request_fiber,
+      remote_file: remote_file
     )
 
-    begin
-      response = request.receive_headers
-      if (URI(uri).scheme != 'file' && response.code != 200)
-        raise RMT::Downloader::Exception.new("#{remote_file} - HTTP request failed with code #{response.code}")
-      end
-    rescue StandardError => e
-      downloaded_file.unlink
-      Fiber.yield # yield, so that on_body callback can be invoked
-      raise e
-    end
+    request.receive_headers
 
     begin
-      response = request.receive_body
-      if (response.return_code && response.return_code != :ok)
-        raise RMT::Downloader::Exception.new("#{remote_file} - return code #{response.return_code}")
-      end
-
-      downloaded_file.close
-
+      request.receive_body
```

Comment:
Why let `#receive_headers` handle its exceptions but not `#receive_body`?

Reply:
I pass the ball to @ikapelyukhin here: :football:

Reply:
I was doing refactoring to fix offenses about method over-complexity. Right now it handles both `#receive_body` and `#verify_checksum` methods exceptions -- but I'll see if I can move `#verify_checksum` into `Request` class as well.

Reply:
Done!